catalog/core/catalog-core-subscriptionstore/src/main/java/org/codice/ddf/catalog/subscriptionstore/ThreadMonitoredSubscriptionCache.java

```diff
@@ -0,0 +1,35 @@
+package org.codice.ddf.catalog.subscriptionstore;
+
+import java.util.Map;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+/**
+ * Temporary measure for refreshing the cache.
+ */
+public class ThreadMonitoredSubscriptionCache extends SubscriptionCache {
+
+    // TODO: Might abstract this into a cache invalidation strategy later
+    private final Executor cacheThread = Executors.newSingleThreadExecutor();
```

Comment:
Depending on how much effort that is, I may just bring in Apache Ignite as part of this change since the threading / scheduling was just a stop-gap for subscription synchronization in an HA environment.